packages/react/scroll-area/src/ScrollArea.tsx

```diff
@@ -747,7 +747,22 @@ interface ScrollAreaThumbProps extends PrimitiveDivProps {
 
 const ScrollAreaThumb = React.forwardRef<ScrollAreaThumbElement, ScrollAreaThumbProps>(
   (props: ScopedProps<ScrollAreaThumbProps>, forwardedRef) => {
-    const { __scopeScrollArea, forceMount, style, ...thumbProps } = props;
+    const { forceMount, __scopeScrollArea, ...contentProps } = props;
+    const scrollbarContext = useScrollbarContext(THUMB_NAME, __scopeScrollArea);
+    return (
+      <Presence present={forceMount || scrollbarContext.hasThumb}>
```

Comment:
Just curious, I took a quick peek at the animated story and couldn't see the exit animation firing. When is `hasThumb` being updated? or is what I see in the story just because the parent scrollbar isn't also animated?

Reply:
The best way to test it in the Content Change story, basically it's when a scrollArea becomes scrollable or non-scrollable, so when content gets added/removed, or it gets resized to the point a thumb isn't necessary (but depending on the `type`, the track might still be displayed).

Reply:
Thanks! yeah looks like we’d need an additional `visible` check or something to support an exit animation as `hasThumb` persists `true`. That would expose other issues though e.g. if the scrollbar is animated but the thumb is not, the thumb would unmount before the scrollbar had finished its exit (which would look a bit janky).

So perhaps it’s fine as is for now.